src/game/interactables_2/091.c

```diff
@@ -0,0 +1,523 @@
+            && (someY + 2 * TILE_WIDTH >= playerY))
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+// static
+void Task_807DE98(void)
+{
+    Sprite_EggUtopia_Launcher *launcher = TaskGetStructPtr(gCurTask);
+
+    if (IS_MULTI_PLAYER) {
+        sub_807E0D0(launcher);
+    }
+
+    if (sub_807DDF0(launcher)) {
+        sub_807DC80(launcher);
+    }
+
+    if (sub_807E044(launcher)) {
+        sub_807E0B8(launcher);
+    } else {
+        sub_807DDA0(launcher);
+    }
+}
+
+static void Task_807DEEC(void)
+{
+    Sprite_EggUtopia_Launcher *launcher = TaskGetStructPtr(gCurTask);
+
+    if (IS_MULTI_PLAYER) {
+        sub_807E0D0(launcher);
+    }
+
+    if (++launcher->unk5C > 60) {
+        SetTaskMain_807E16C(launcher);
+    }
+
+    sub_807DDA0(launcher);
+}
+
+// static
+void TaskDestructor_807DF38(struct Task *t)
+{
+    Sprite_EggUtopia_Launcher *launcher = TaskGetStructPtr(t);
+    VramFree(launcher->s.graphics.dest);
+}
+
+static void SetTaskMain_807E16C(Sprite_EggUtopia_Launcher *unused)
+{
+    gCurTask->main = Task_807E16C;
+}
+
+static bool16 sub_807DF60(Sprite_EggUtopia_Launcher *launcher)
+{
+    bool32 result = FALSE;
+
+    if (IS_LAUNCHER_DIR_LEFT(launcher->kind)) {
+        s32 someX;
+        launcher->unk54 -= Q_24_8(15);
+        someX = Q_24_8(launcher->posX + launcher->unk4C);
+        if (launcher->unk54 <= someX) {
+            launcher->unk54 = someX;
+            result = TRUE;
+        }
+    } else {
+        s32 someX;
+        launcher->unk54 += Q_24_8(15);
+        someX = Q_24_8(launcher->posX + launcher->unk50);
+        if (launcher->unk54 >= someX) {
+            launcher->unk54 = someX;
+            result = TRUE;
+        }
+    }
+
+    return result;
+}
+
+static void sub_807DFBC(Sprite_EggUtopia_Launcher *launcher)
+{
+    if (PLAYER_IS_ALIVE && launcher->unk48) {
+        switch (launcher->kind) {
+            case LAUNCHER_KIND(LAUN_DIR_LEFT, LAUN_GRAVITY_DOWN): {
+                gPlayer.x = launcher->unk54 - Q_24_8(8);
+                gPlayer.y = launcher->unk58 - Q_24_8(16);
+            } break;
+
+            case LAUNCHER_KIND(LAUN_DIR_RIGHT, LAUN_GRAVITY_DOWN): {
+                gPlayer.x = launcher->unk54 + Q_24_8(8);
+                gPlayer.y = launcher->unk58 - Q_24_8(16);
+            } break;
+
+            case LAUNCHER_KIND(LAUN_DIR_LEFT, LAUN_GRAVITY_UP): {
+                gPlayer.x = launcher->unk54 - Q_24_8(8);
+                gPlayer.y = launcher->unk58 + Q_24_8(16);
+            } break;
+
+            case LAUNCHER_KIND(LAUN_DIR_RIGHT, LAUN_GRAVITY_UP): {
+                gPlayer.x = launcher->unk54 + Q_24_8(8);
+                gPlayer.y = launcher->unk58 + Q_24_8(16);
+            } break;
+        }
+    }
+}
+
+static bool32 sub_807E044(Sprite_EggUtopia_Launcher *launcher)
+{
+    s16 posX, posY;
+
+    posX = launcher->posX - gCamera.x;
+    posY = launcher->posY - gCamera.y;
+
+    if (((posX + launcher->unk50) < -(CAM_REGION_WIDTH / 2))
```

Comment:
Does this fit any macros?

Reply:
Not *yet*. IS_OUT_OF_CAM_RANGE in game.h is closest, but we'd need a version supports top, bottom, left and right.
But in general they all have to be cleaned up and (hopefully) turned into a single or very few macros.

Reply:
For the record, that's also the case for small_propeller!